apypie/action.py

```diff
@@ -227,3 +230,14 @@ def _prepare_params(self, action_params, input_dict):
                 result[param.name] = input_dict[param.name]
 
         return result
+
+    @staticmethod
+    def _prepare_route_params(route, input_dict):
+        # type: (Route, dict) -> dict
+        result = {}
+
+        for url_param in route.params_in_path:
+            if url_param in input_dict:
```

Comment:
Should a missing `url_param` not be considered an error?

Reply:
Technically, yes.
But this is getting called with `action.find_best_route(params)`, so this should never happen.

Reply:
But maybe the call to `find_best_route` should be explicit *inside* this function?

Reply:
(updated)

Reply:
:+1:

So `find_best_route` guarantees that all `params_in_path` are available? Then we really don't need the extra `if`.

Reply:
Yes, it picks the route that consumes as many params as possible.

Reply:
Turns out, it was needed *scratch head*

Reply:
?? So there are optional url_parameter? I'm not going to block on this here, though.

Reply:
FAM tests failed when I dropped the if, and I am not going to find out why now ;)